apm-sniffer/optional-plugins/pom.xml

```diff
@@ -98,11 +102,22 @@
                         <configuration>
                             <tasks>
                                 <taskdef resource="net/sf/antcontrib/antcontrib.properties" classpathref="maven.runtime.classpath" />
+                                <mkdir dir="${optional.agent.package.dest.dir}" />
                                 <if>
                                     <equals arg1="${project.packaging}" arg2="jar" />
                                     <then>
-                                        <mkdir dir="${plugin.dest.dir}" />
-                                        <copy file="${project.build.directory}/${project.artifactId}-${project.version}.jar" tofile="${plugin.dest.dir}/${project.artifactId}-${project.version}.jar" overwrite="true" />
+                                        <if>
+                                            <!-- if the plugin is a directory -->
```

Comment:
I am wondering, whether be better if put conf and jar in the `optional-plugins` dictionary directly? I feel strange about this part. Or do you have any suggestion?

Reply:
My mistake, i'll remove that

Reply:
Well, I think the meaning of this plugin may be somewhat confusing,
One is to integrate other components to achieve link tracking.
The other is to extend SPI's own behavior.

Reply:
If there is a standard example of an extension to SPI, it may be clear.

Reply:
I am following. I just have concerns about this pom. Too many your plugin details expose to root pom.